ydb/core/blobstorage/dsproxy/request_history.h

```diff
@@ -0,0 +1,206 @@
+#pragma once
+#include "defs.h"
+
+#include <variant>
+
+namespace NKikimr {
+class THistory {
+private:
+    struct TBaseEntry {
+        TDuration Timestamp;
+        TBaseEntry(TInstant startTime)
+            : Timestamp(TInstant::Now() - startTime)
+        {}
+    };
+
+    struct TVRequestEntry : public TBaseEntry {
+        ui8 PartIdx;
```

Comment:
Now if you really want to save space, you can move ui8's together, so they don't lose so much on misalign.